include/aspect/material_model/visco_plastic.h

```diff
@@ -168,11 +168,36 @@ namespace aspect
 
         double grain_size;
 
-        std::vector<double> densities;
+        // list of depth (or pressure), width and Clapeyron slopes
+        // for the different phase transitions
+        std::vector<double> transition_depths;
+        std::vector<double> transition_pressures;
+        std::vector<double> transition_temperatures;
+        std::vector<double> transition_widths;
+        std::vector<double> transition_pressure_widths;
+        std::vector<double> transition_slopes;
+
+        // Number of phase changes
+        unsigned int number_of_phase_transitions;
+
+        // Use depth (if true) instead of pressure to define phase transitions
+        bool use_depth;
+
+        dealii::Table<2,double> densities;
```

Comment:
You should document what the rows and columns of the table are (compositions and phases).